src/main.js

```diff
@@ -7,6 +7,10 @@ npath = require('./path');
 
 child_process = require('child_process');
 
+function encodeNonBuffer(data, encoding) {
```

Comment:
Won't it better to put this function into the `src/utils.js` file?

Reply:
Probably would be better. I'll update the PR.